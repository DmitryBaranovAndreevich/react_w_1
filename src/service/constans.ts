export const apiConfig = {
  baseUrl: 'https://moviesdatabase.p.rapidapi.com',
  headers: {
    'X-RapidAPI-Key': 'd878733499msh4f2be08f007b5e6p1f0753jsndfa44e7b9e5b',
    'X-RapidAPI-Host': 'moviesdatabase.p.rapidapi.com',
  },
};
