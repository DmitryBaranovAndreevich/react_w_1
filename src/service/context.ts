import IItem from 'interfaces/IItem';
import React, { Dispatch, SetStateAction } from 'react';
const init: [Array<IItem>, Dispatch<SetStateAction<IItem[]>>] = [[], () => {}];
export const Context = React.createContext(init);
