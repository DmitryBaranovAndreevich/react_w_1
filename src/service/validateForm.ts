import TDataForm from '../interfaces/TDataForm';

function validationForm({ userName, surName, date, select, checkBox, file }: TDataForm) {
  const validateFirstLetterToUpperCase = (str: string) =>
    !str ? true : !(str[0].toUpperCase() === str[0]);

  const validateDate = (date: string) => !date;

  const validateCheckBox = (value: boolean) => !value;

  const validateSelect = (value: string) => (value === 'default' ? true : false);

  const validateFile = (value: string) => !value;

  return {
    validName: validateFirstLetterToUpperCase(userName),
    validSurName: validateFirstLetterToUpperCase(surName),
    validDate: validateDate(date),
    validSelect: validateSelect(select),
    validCheckBox: validateCheckBox(checkBox),
    validFile: validateFile(file),
  };
}

export default validationForm;
