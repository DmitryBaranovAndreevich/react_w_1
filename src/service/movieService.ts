import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/dist/query/react';
import { apiConfig } from './constans';
import IItem from 'interfaces/IItem';

interface Responce {
  page: number;
  next: string;
  entries: number;
  results: IItem[];
}

export const moviesApi = createApi({
  reducerPath: 'moviesApi',
  baseQuery: fetchBaseQuery({ baseUrl: apiConfig.baseUrl }),
  endpoints: (build) => ({
    fetchAllMovies: build.query<Responce, string>({
      query: () => ({
        url: '/titles',
        headers: apiConfig.headers,
      }),
    }),

    fetchSearchMovies: build.query<Responce, string>({
      query: (url) => ({
        url: `/titles/search/title/${url}`,
        headers: apiConfig.headers,
      }),
    }),
  }),
});
