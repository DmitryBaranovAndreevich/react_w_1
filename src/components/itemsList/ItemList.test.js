import ItemsList from './itemsList';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { goods } from '../../service/goods';

describe('Тест компонента списка товаров', () => {
  it('Компонент списка товаров рендерится без проблем с пустым списком', () => {
    const searchBar = renderer.create(<ItemsList goodsList={[]} />).toJSON();
    expect(searchBar).toMatchSnapshot();
  });

  it('Компонент списка товаров рендерится без проблем со списком товаров', () => {
    const { container } = render(
      <BrowserRouter>
        <ItemsList goodsList={goods.products} />
      </BrowserRouter>
    );

    const items = container.firstChild.childNodes;
    expect(items.length).toBe(goods.products.length);
  });
});
