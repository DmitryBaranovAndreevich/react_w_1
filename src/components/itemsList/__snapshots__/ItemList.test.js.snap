// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента списка товаров Компонент списка товаров рендерится без проблем с пустым списком 1`] = `
<div
  className="container"
/>
`;

exports[`Тест компонента списка товаров Компонент списка товаров рендерится без проблем с пустым списком 2`] = `
<div
  className="container"
/>
`;

exports[`Тест компонента списка товаров Компонент списка товаров рендерится без проблем сj cgbcrjv  1`] = `
<div
  className="container"
/>
`;

exports[`Тест компонента списка товаров Компонент списка товаров рендерится без проблем со списком товаров 1`] = `
<div
  className="container"
/>
`;
